<?xml version="1.0"?>
<launch>
    <arg name="sim" default="0"/>

  <include file="$(find crazyflie_driver)/launch/crazyflie_server.launch"></include>

   <include file="$(find gmodrotor_control)/launch/gcrazyflie.launch">
      <arg name="model" value="crazyflie01"/>
      <arg name="uri" value="radio://0/35/2M" />
      <arg name="sim" value="$(arg sim)"/>
  </include>

  <include file="$(find gmodrotor_control)/launch/gcrazyflie.launch">
      <arg name="model" value="crazyflie02"/>
      <arg name="uri" value="radio://0/32/2M" />
      <arg name="sim" value="$(arg sim)"/>
  </include>

  <include file="$(find gmodrotor_control)/launch/gcrazyflie.launch">
      <arg name="model" value="crazyflie03"/>
      <arg name="uri" value="radio://0/33/2M" />
      <arg name="sim" value="$(arg sim)"/>
  </include>

  <include file="$(find gmodrotor_control)/launch/gcrazyflie.launch">
      <arg name="model" value="crazyflie04"/>
      <arg name="uri" value="radio://0/38/2M" />
      <arg name="sim" value="$(arg sim)"/>
  </include>




  <node pkg="gmodrotor_control" type="pba.py" name="pba"/>

  <node pkg="mocap_vicon"
    type="mocap_vicon_node"
    name="vicon"
    output="screen" unless="$(arg sim)">
    <param name="server_address" value="192.168.129.201"/>
    <param name="server_base_port" value="22222"/>
    <param name="frame_rate" value="100"/>
    <param name="max_accel" value="10.0"/>
    <param name="publish_tf" value="true"/>
    <param name="fixed_frame_id" value="mocap"/>
    <rosparam param="model_list">[crazy02, crazy03, crazy05, crazy08]</rosparam>

    <remap from="vicon/crazy02/odom" to="/crazyflie02/odom"/>
    <remap from="vicon/crazy08/odom" to="/crazyflie04/odom"/>
    <remap from="vicon/crazy05/odom" to="/crazyflie01/odom"/>
    <remap from="vicon/crazy03/odom" to="/crazyflie03/odom"/>
  </node>


</launch>
